src/services/web-monitoring-api.js

```diff
@@ -16,76 +16,6 @@ export default class WebMonitoringApi {
     this.dbApi = dbApi;
   }
 
-  /**
-     * Get the current analysis timeframe a user should be working on.
-     * @returns {Promise<AnalysisTimeframe>}
-     */
-  getCurrentTimeframe () {
-    return fetch(`/api/timeframe?date=${new Date().toISOString()}`)
-      .then(response => response.json())
-      .then(timeframe => {
-        if (timeframe.error) {
-          throw new Error(timeframe.error);
-        }
-
-        return {
-          duration: timeframe.duration,
-          end: new Date(timeframe.end),
-          start: new Date(timeframe.start)
-        };
-      });
-  }
-
-  /**
-     * Get a list of domains assigned to a given user.
-     * @param {string} username
-     * @returns {Promise<string[]>}
-     */
-  getDomainsForUser (username) {
-    if (!username) {
-      return Promise.reject(new TypeError('The first argument to getDomainsForUser() must be a string.'));
-    }
-
-    const url = `/api/domains/${username}`;
-    return fetch(url)
-      .then(response => response.json())
-      .then(data => {
-        if (data.error) {
-          throw new Error(data.error);
-        }
-
-        return data.domains;
-      });
-  }
-
-  /**
-     * Get a list of pages a user should analyze based on their assigned
-     * domains and the analysis timeframe.
-     *
-     * This promise can resolve to an array of pages OR to `null`, which
-     * indicates the user has no assigned pages to monitor (an empty array of
-     * pages means none of their pages were updated in the given timeframe).
-     *
-     * @param {string} username
-     * @param {AnalysisTimeframe} [timeframe] If omitted, defaults to current
-     * @param {object} [query] Additional query options for getting pages
-     * @returns {Promise<Page[]>}
-     */
-  getPagesForUser (username, timeframe, query = {}) {
-    const domainsRequest = this.getDomainsForUser(username);
-    const timeframeRequest = Promise.resolve(timeframe || this.getCurrentTimeframe());
-
-    return Promise.all([domainsRequest, timeframeRequest])
-      .then(([domains, timeframe]) => {
-        if (domains.length === 0) return null;
-        return this._getPagesByDomains(
```

Comment:
This is the only place `_getPagesByDomains` is used, so you can probably remove that function as well.